### DISCUSSION TITLE
Probe

### DISCUSSION THREAD

**Discussion by kotovasia5120** [kachurovskiy/nanoels#204] [Q&A]
Can I use a feeler gauge to indicate the axis zero?
[For example this one.](https://sl.aliexpress.ru/p?key=tsQ2sIy)

**Comment by kachurovskiy** [comment]
Hi @kotovasia5120, I can't open the link, perhaps it's geo-limited. I'm assuming you're talking about a fixed-height button-style probe with internal spring.

I was thinking about ways to use them for zeroing the X axis but never got to that. There will be some challenges - how do you position the probe perfectly vertical between the tool and the part? If it's at an angle, the measurement will be off.

If you have a spare port on the back of H4, you could try to make some code changes to make it work, please let us know the results.

**Reply by kotovasia5120** [reply]
No, I sent a probe based on the electric current between the sensor and the tip of the tool.
![Screenshot_20240711-111732](https://github.com/kachurovskiy/nanoels/assets/76496357/343cebfe-124f-47ec-b755-6609eeb982bb)
![Screenshot_20240711-111803](https://github.com/kachurovskiy/nanoels/assets/76496357/5c7eda44-b0cb-41fd-9c35-99379254a427)

**Reply by kachurovskiy** [reply]
Yeah, I was thinking about the same tool. Let us know if you can make it work.

**Reply by Festivejelly** [reply]
Could use the same solution as using a limit switch.

For the X axis you would need 2 mounts for this. Because you would want to zero a boring bar and also a turning/facing tool.

Could maybe use a 3d printed mount to mount it to the splashguard for setting the touch off for Facing tools.

Then maybe have  3d printed mount for it build into the X axis cross slide motor mount.

Though I think a better way to do it is have it mounted in 1 place (splashguard) Then create an offsets table for the different tools like i've done in my branch:

https://github.com/Festivejelly/nanoels/blob/4c66b5fd97d285aca0d66333d9eb860bd44e0340/h4/h4.ino#L306

With this you have a primary tool which you zero.

Then you build a library of tools that are offset from this primary tool. It works well for me and they seem to hold their zero well on a quick change tool post.

**Comment by Festivejelly** [comment]
I chuck one of these up in the 3 jaw or collet chuck. Then I move the tool to the middle of the sphere and when it beeps just put 10 then press A (as the ball bearing diameter is 10mm)

I found these to be very accurate.

https://www.amazon.co.uk/GXMZL-Precision-Electronic-Milling-Machine/dp/B081CMY91B/ref=sr_1_18?dib=eyJ2IjoiMSJ9.3oD86PtiuWX0rAupoPSKAjR-_eQxsZSgEwsyxjLDvvbBOb2KcVd8PLFgOn4CdjRzIYy3Yw6WsSlfdrv9affqKEzPQLZegq8tQJFnqu4unZUS6JdPDtZjv5AXX8D7Fymxm6OP9Yyz4pgluaPA7q8wzlSz0TUfcQMo7XFGmmW84oZ4jrJGR4SgRX6jxthFClM_BxI2eWKU5TgHIl70JFVOrKkqWJNv9DlK1MLJrBYbalGW7A_RIjHyaap28wT_vIj1UaBqjzAnKdrLKYn54oUwsUkjkVhJarIIOqE87FodL7I.vWMOKgx6zOwOP-pRq4hwll_XcifuP3H9C1Jiubb5Awo&dib_tag=se&keywords=cnc+probe&qid=1724877402&sr=8-18
